zanstat/Program.cs

```diff
@@ -1,104 +1,153 @@
 ﻿using System;
+using Zanlib;
 
 namespace Zanstat
 {
     class Program
     {
-        static void Main(string[] args)
+        static string hostname = "localhost";
+        static int port = 10666;
+
+        static int Main(string[] args)
         {
-            Zanlib.Huffman.Init();
+            try
+            {
+               
 
-            var hostname = "localhost";
-            var port = 10666;
+                return new Program().QueryZandronumServer(args);
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e);
+                return 1;
+            }
+        }
 
+        public int QueryZandronumServer(string[] args)
+        {
             for (var i = 0; i < args.Length; i++)
             {
                 if (args[i] == "-runtests")
+                    return RunTests();
+
+                if (args[i] == "-server")
                 {
-                    Tests.TestUncompressed();
-                    Tests.TestCompressed();
-                    Console.ReadKey();
-                    return;
-                }
-                else if (args[i] == "-server")
-                {
-                    if (i < args.Length)
-                    {
-                        var server = args[i + 1];
-                        var colon = server.IndexOf(':');
-                        if (colon >= 0)
-                        {
-                            hostname = server.Substring(0, colon);
-                            port = int.Parse(server.Substring(colon + 1));
-                        }
-                        else
-                        {
-                            hostname = server;
-                        }
-                    }
-                    else
-                    {
-                        Console.WriteLine("Must specify a server!");
-                        return;
-                    }
+                    if (!GetServerArg(i, args))
+                        return 1;
                 }
             }
 
-            Console.WriteLine($"Connecting to {hostname}:{port}");
+            Console.WriteLine($"Connecting to {hostname}:{port} ... ");
+
+            ZandronumRcon Zanlib = new ZandronumRcon(hostname, port);
+
             for (var i = 0; i < args.Length; i++)
             {
                 if (args[i] == "-getname")
                 {
-                    var serverName = Zanlib.Name.Get(hostname, port);
+                    var serverName = Zanlib.Name.Get();
```

Comment:
Actually this is one nice thing of object-ifying the code - no need to keep passing in the hostname and port. I'm still unconvinced though!